### PR TITLE
fix(tickets block): Fixes duplicate block on posts/pages.

### PR DESCRIPTION
**Pull request by Camwyn**
Also Fixes:
Missing js vars on posts/pages.
Fixes BEM of notice classes.
Moves notice to a honest-to-goodness template!
Adds a style override for `twentyseventeen` (dialog z-index).
🎫 https://central.tri.be/issues/132568
🎫 https://central.tri.be/issues/134557

Added AR page Design QA as I need some of this new structure to implement.

### CODE REVIEW

File: src/views/registration-js/content.php [hunk 1]
```diff
@@ -143,24 +146,29 @@ class="tribe-tickets__item__attendee__fields__form <?php echo sanitize_html_clas
...
+		<?php
+		$notice_classes = [
+			'tribe-tickets__notice--non-ar',
+		];
+
+		if ( ! empty( $non_meta_count ) ) {
+			$notice_classes[] = 'tribe-common-a11y-hidden';
+		}
+
+		tribe( 'tickets.editor.template' )->template(
```
Comment (sc0ttkclark) [inline src/views/registration-js/content.php]: Should this be:

```suggestion
		$this->template(
```

File: src/views/registration-js/content.php [hunk 2]
```diff
@@ -39,42 +39,45 @@
 	<div class="tribe-common-h8 tribe-common-h--alt tribe-tickets__registration__actions">
 		<?php $this->template( 'registration/button-cart', array( 'provider' => $provider ) ); ?>
 	</div>
-	<h1 class="tribe-common-h2 tribe-common-h3--min-medium tribe-common-h--alt tribe-tickets__registration__page-title"><?php esc_html_e( 'Attendee Registration', 'event-tickets'); ?></h1>
+	<h1 class="tribe-common-h2 tribe-common-h1--min-medium tribe-common-h--alt tribe-tickets__registration__page-title"><?php esc_html_e( 'Attendee Registration', 'event-tickets'); ?></h1>
```
Comment (sc0ttkclark) [inline src/views/registration-js/content.php]:
```suggestion
	<h1 class="tribe-common-h2 tribe-common-h1--min-medium tribe-common-h--alt tribe-tickets__registration__page-title"><?php esc_html_e( 'Attendee Registration', 'event-tickets' ); ?></h1>
```

File: src/views/modal/registration-js.php [hunk 3]
```diff
@@ -62,23 +66,30 @@ class="tribe-tickets__item__attendee__fields__form <?php echo sanitize_html_clas
...
+		<?php
+		$notice_classes = [
+			'tribe-tickets__notice--non-ar',
+		];
+
+		if ( ! empty( $non_meta_count ) ) {
+			$notice_classes[] = 'tribe-common-a11y-hidden';
+		}
+
+		tribe( 'tickets.editor.template' )->template(
```
Comment (sc0ttkclark) [inline src/views/modal/registration-js.php]: Should this be:

```suggestion
		$this->template(
```

File: src/views/modal/registration-js.php [hunk 4]
```diff
@@ -23,21 +23,25 @@
...
-						'You have %s ticket(s) with a field that requires information.',
-						'Note about missing required fields, %s is the html-wrapped number of tickets.',
-						'event-tickets'
-					),
-					'<span class="tribe-tickets-notice--error__count">1</span>'
-				);
-			?>
-		</p>
-	</div>
+	<?php tribe( 'tickets.editor.template' )->template(
```
Comment (sc0ttkclark) [inline src/views/modal/registration-js.php]: Should this be:

```suggestion
		$this->template(
```

File: src/views/modal/registration-js.php [hunk 5]
```diff
@@ -10,20 +10,24 @@
...
+$provider_class        = $view->get_form_class( $providers_arr[0] );
 $has_tpp               = in_array( Tribe__Tickets__Commerce__PayPal__Main::ATTENDEE_OBJECT, $providers, true );
 $event_id              = get_the_ID();
 $meta                  = Tribe__Tickets_Plus__Main::instance()->meta();
 $non_meta_count        = 0;
 ?>
 <div class="tribe-tickets__item__attendee__fields">
 	<h2 class="tribe-common-h3 tribe-common-h4--min-medium tribe-common-h--alt tribe-tickets__item__attendee__fields__title"><?php esc_html_e( 'Attendee Details', 'event-tickets' ); ?></h2>
-	<?php tribe( 'tickets.editor.template' )->template(
+	<?php $template->template(
```
Comment (sc0ttkclark) [inline src/views/modal/registration-js.php]: `$this`?

File: src/views/modal/registration-js.php [hunk 6]
```diff
@@ -76,7 +80,7 @@ class="tribe-tickets__item__attendee__fields__form <?php echo sanitize_html_clas
 			$notice_classes[] = 'tribe-common-a11y-hidden';
 		}
 
-		tribe( 'tickets.editor.template' )->template(
+		$template->template(
```
Comment (sc0ttkclark) [inline src/views/modal/registration-js.php]: `$this`?

File: src/views/modal/registration-js.php (hunk 5 above)
Comment (Camwyn) [reply src/views/modal/registration-js.php]: nope - this gets used in the modal and the modal doesn't have the template function carry through.

File: src/views/modal/registration-js.php (hunk 6 above)
Comment (Camwyn) [reply src/views/modal/registration-js.php]: nope - this gets used in the modal and the modal doesn't have the template function carry through.

File: src/views/modal/registration-js.php [hunk 7]
```diff
@@ -62,23 +70,30 @@ class="tribe-tickets__item__attendee__fields__form <?php echo sanitize_html_clas
...
 						'Note that there are more tickets in the cart, %s is the html-wrapped number.',
 						'event-tickets'
 					),
 					'<span id="tribe-tickets__non-ar-count">' . absint( $non_meta_count ) . '</span>'
-				);
-			?>
-		</p>
+				)
+			]
+		); ?>
```
Comment (sc0ttkclark) [inline src/views/modal/registration-js.php]:
```suggestion
		);
		?>
```

**Pull request merged**